btk/metrics.py

```diff
@@ -1,484 +1,279 @@
-        )
-
-    num_matched_detections2 = true_table["num_detections2"].sum()
-    if not num_matched_detections2 + num_spurious2 == num_det:
-        raise ValueError(
-            "Number of detections match to a true object + "
-            "number of spurious must be equal to the total "
-            "number of detections."
-        )
-
-    blend_summary = [
-        num_true,
-        num_detected1,
-        num_undetected1,
-        num_spurious1,
-        num_shred1,
-        num_detected2,
-        num_undetected2,
-        num_spurious2,
-        num_shred2,
-    ]
-    return blend_summary
+    # for each true galaxy i, match_indx[i] is the index of detected_table matched to that true
+    # galaxy or -1 if there is no match.
+    match_indx = [-1] * len(true_table)
+    dist_m = [0.0] * len(true_table)
+    for i, indx in enumerate(true_indx):
+        match_indx[indx] = detected_indx[i]
+        dist_m[indx] = dist[indx][detected_indx[i]]
 
+    match_table["match_detected_id"] = match_indx
+    match_table["dist"] = dist_m
+    return match_table
 
-def get_detection_eff_matrix(summary_table, num):
-    """Computes the detection efficiency matrix for the input detection summary
-    table.
 
-    Input argument num sets the maximum number of true objects per blend in the
-    test set for which the
-    detection efficiency matrix is to be created for. Detection efficiency is
-    computed for a number of true objects in the range (0-num) as columns and
-    the detection percentage as rows. The percentage values in a column sum to
-    100.
-
-    The input summary table must be a numpy array of shape [N, 5], where N is
-    the test set size. The 5 columns in the summary_table are number of true
-    objects, detected sources, undetected objects, spurious detections and
-    shredded objects for each of the N blend scenes in the test set.
-
-    Args:
-        summary_table (`numpy.array`): Detection summary as a table [N, 5].
-        num (int): Maximum number of true objects to create matrix for. Number
-            of columns in efficiency matrix will be num+1. The first column
-            will correspond to no true objects.
-
-    Returns:
-        numpy.ndarray of size[num+2, num+1] that shows detection efficiency.
-    """
-    eff_matrix = np.zeros((num + 2, num + 1))
-    for i in range(0, num + 1):
-        (q_true,) = np.where(summary_table[:, 0] == i)
-        for j in range(0, num + 2):
-            if len(q_true) > 0:
-                (q_det,) = np.where(summary_table[q_true, 1] == j)
-                eff_matrix[j, i] = len(q_det)
-    norm = np.sum(eff_matrix, axis=0)
-    # If no detections along a column, set sum to 1 to avoid dividing by zero.
-    norm[norm == 0.0] = 1
-    # normalize over columns.
-    eff_matrix = eff_matrix / norm[np.newaxis, :] * 100.0
-    return eff_matrix
+def detection_metrics(blended_images, isolated_images, blend_list, detection_catalogs, matches):
```

Comment:
I noticed some of these arguments are unused, is that intentional?

Reply:
Kind of, I just included all the possibly useful arguments in the functions ; I will probably remove them in the future.